website/docs/language/settings/backends/configuration.mdx

```diff
@@ -35,6 +43,18 @@ There are some important limitations on backend configuration:
 - A configuration can only provide one backend block.
 - A backend block cannot refer to named values (like input variables, locals, or data source attributes).
 
+### Credentials and Sensitive Data
+
+Backends store state in a remote service, which allows multiple people to access it. Accessing remote state generally requires access credentials, since state data contains extremely sensitive information.
+
+!> **Warning:**  We recommend using environment variables to supply credentials and other sensitive data. If you use `-backend-config` or hardcode these values directly in your configuration, Terraform will include these values in both state and plan files. This can leak sensitive credentials.
```

Comment:
I broadly like this way of presenting it, but it isn't technically correct to refer to `.terraform/terraform.tfstate` as a "state file" in the usual sense we mean when we say that. I don't know that we really have a good name for that file because we haven't really needed to talk about it in the documentation before.

What do you think about this alternative for the last two sentences?

> If you use `-backend-config` or hardcode these values directly in your configuration, Terraform will include these values in both the `.terraform` subdirectory and in plan files. This can leak sensitive credentials.

Here I referred to the `.terraform` directory broadly, since I expect that's something users will either already be familiar with or will become accustomed to quickly when using Terraform, whereas the exact contents of that directory are not so relevant to everyday Terraform use because `terraform init` (along with some other commands) manages it all automatically.

I think it's fine to refer to the specific path `.terraform/terraform.tfstate` below where we have room for more words to say a little about what that file is, but I don't think it's necessary to go into that much detail in the little callout box we're going to copy onto every backend page.

Reply:
I like it! Let me push up a change where I fix all the little boxes with this nice new sentence. And yeah that makes sense to just give the general warning without going into very specific details in the callout. Thank you very much for explaining!